arch/riscv/build.mk

```diff
@@ -17,27 +17,49 @@ DEFINES := -DF_CPU=$(F_CLK) \
            -DF_TIMER=$(F_TICK) \
            -include config.h
 
-ASFLAGS = -march=rv32imzicsr -mabi=ilp32
-CFLAGS += -Wall -Wextra -Wshadow -Wno-unused-parameter -Werror
+CROSS_COMPILE ?= riscv32-unknown-elf-
+CC_DEFAULT := $(CROSS_COMPILE)gcc
+CC_IS_CLANG := $(shell $(CC_DEFAULT) --version 2>/dev/null | grep -qi clang && echo 1)
```

Comment:
I find this approach a bit odd. Why append the gcc suffix and then check for clang? It feels a bit roundabout.